### PR TITLE
Allow axes to be hidden via an orientation of 'none'

### PR DESCRIPTION
**Pull request by ColinEberhardt**
closes https://github.com/d3fc/d3fc/issues/1061

This, plus the addition of making these properties functors, is a useful feature for creating small multiple charts.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,10 +91,17 @@ For charts that contain a very high number of data-points, rendering to canvas c
 
 If *label* is specified, sets the text for the given label, and returns the Cartesian chart. If *label* is not specified, returns the label text.
 
+The *label* value can either be a string, or a function that returns a string. If it is a function, it will be invoked with the data that is 'bound' to the chart. This can be useful if you are rendering multiple charts, with different chart labels, using a data join.
+
 <a name="cartesian_xOrient" href="#cartesian_xOrient">#</a> *cartesian*.**xOrient**(*orient*)  
 <a name="cartesian_yOrient" href="#cartesian_yOrient">#</a> *cartesian*.**yOrient**(*orient*)  
 
-If *orient* is specified, sets the orientation for the axis in the given direction, and returns the Cartesian chart. If *orient* is not specified, returns the orientation. Valid values for *yOrient* are *left* or *right*, and for *xOrient*, *top* or *bottom*.
+If *orient* is specified, sets the orientation for the axis in the given direction, and returns the Cartesian chart. If *orient* is not specified, returns the orientation. Valid values for *yOrient* are *left*, *right* or *none*, and for *xOrient* they are *top*, *bottom* or *none*.
```
Comment (chrisprice) [inline README.md]: Am I right in thinking that orient can't change between invocations? 
It feels weird that previously it was a property and not a "constructor argument". Not sure making it a functor changes much but it still feels odd...

File: src/cartesianBase.js [hunk 2]
```diff
@@ -80,45 +102,49 @@ export default (d3fcElementType, plotAreaDrawFunction) =>
                 container.select('.chart-label')
                     .text(chartLabel(data));
 
-                container.select('.y-axis')
-                    .on('measure', (d, i, nodes) => {
-                        if (yOrient === 'left') {
-                            const { width, height } = event.detail;
-                            select(nodes[i])
+                if (yAxis) {
```
Comment (chrisprice) [inline src/cartesianBase.js]: I don't think you need the guard, empty selections should just no-op over the subsequent calls.

File: src/cartesianBase.js [hunk 3]
```diff
@@ -80,45 +102,49 @@ export default (d3fcElementType, plotAreaDrawFunction) =>
...
+                              .call(yAxis.scale(yScale));
+                        });
+                }
 
-                container.select('.x-axis')
-                    .on('measure', (d, i, nodes) => {
-                        if (xOrient === 'top') {
-                            const { width, height } = event.detail;
-                            select(nodes[i])
+                if (xAxis) {
```
Comment (chrisprice) [inline src/cartesianBase.js]: Same as above

File: README.md (hunk 1 above)
Comment (ColinEberhardt) [inline README.md]: Very good point - I was going to raise an issue on that point - there are already a few places where changes in property values will not have the expected behaviour. My plan was to re-visit this issue in the future.

Comment (ColinEberhardt) [inline README.md]: see #22

**Pull request merged**